301d58/code-challenges/chaining/challenges-10.test.js

```diff
@@ -0,0 +1,214 @@
+
+You may want to use filter, map, or reduce for this problem, but are not required to. You may need to use the same method more than once.
+
+For example, [[1, 2, 3, 4, 5], [6, 7, 2, 4, 5, 7], [9, 2, 3, 6,]] returns 66.
+------------------------------------------------------------------------------------------------ */
+
+const totalSum = (input) => {
+  let sum = 0;
+  input.map( arr => {
+    arr.map( value => {
+      sum = value + sum;
+    });
+    return sum;
+  });
+  return sum;
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 3
+
+Write a function named divisibleByFiveTwoToThePower that accepts an array of arrays as input.
+
+This function should first remove any elements that are not numbers or are not divisible by five.
+
+This function should then raise 2 to the power of the resulting numbers, returning an array of arrays.
+
+For example, [ [0,2,5,4], [2,4,10], [] ] should return [ [1, 32], [1024], [] ].
+------------------------------------------------------------------------------------------------ */
+
+const divisibleByFiveTwoToThePower = (input) => {
+    return input.map( arr => {
+    return arr.filter(value => {
+      return (typeof value === 'number' && value % 5 === 0);
+    }).map( value => {
+      return Math.pow(2, value);
+    });
+  });
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 4
+
+Write a function named findMaleAndFemale that, given the Star Wars data, below,
+returns the names of the characters whose gender is either male or female.
+
+The names should be combined into a single string with each character name separated by "and".
+
+For example, "C-3PO and Luke Skywalker".
+------------------------------------------------------------------------------------------------ */
+
+let starWarsData = [{
+  name: 'Luke Skywalker',
+  height: '172',
+  mass: '77',
+  hair_color: 'blond',
+  skin_color: 'fair',
+  eye_color: 'blue',
+  birth_year: '19BBY',
+  gender: 'male',
+},
+{
+  name: 'C-3PO',
+  height: '167',
+  mass: '75',
+  hair_color: 'n/a',
+  skin_color: 'gold',
+  eye_color: 'yellow',
+  birth_year: '112BBY',
+  gender: 'n/a'
+},
+{
+  name: 'R2-D2',
+  height: '96',
+  mass: '32',
+  hair_color: 'n/a',
+  skin_color: 'white, blue',
+  eye_color: 'red',
+  birth_year: '33BBY',
+  gender: 'n/a'
+},
+{
+  name: 'Darth Vader',
+  height: '202',
+  mass: '136',
+  hair_color: 'none',
+  skin_color: 'white',
+  eye_color: 'yellow',
+  birth_year: '41.9BBY',
+  gender: 'male'
+},
+{
+  name: 'Leia Organa',
+  height: '150',
+  mass: '49',
+  hair_color: 'brown',
+  skin_color: 'light',
+  eye_color: 'brown',
+  birth_year: '19BBY',
+  gender: 'female'
+}];
+
+let findMaleAndFemale = (data) => {
+  let newArr = [];
+  data.map( obj => {
+    if(obj.gender === 'male' || obj.gender === 'female'){
+      newArr.push(obj.name);
```

Comment:
Because you're using `.map` you could have returned `obj.name` instead of pushing it into `newArr` (and then you wouldn't have needed to create `newArr`)